tests/unit_tests/test_dqc_us_0046.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from arelle.ModelManager import ModelManager
+from arelle.ModelXbrl import ModelXbrl
+from arelle.ModelRelationshipSet import ModelRelationshipSet
+from arelle import ValidateXbrl, ModelDtsObject
+from dqc_us_rules import dqc_us_0046
+
+
+class TestDQC0046(unittest.TestCase):
+    def setUp(self):
+        self.concept_1 = MagicMock(
+            spec=ModelDtsObject.ModelConcept,
+            qname=MagicMock(
+                namespaceURI='http://fasb.org/us-gaap/2015-01-31',
+                localName='Assets'
+            )
+        )
+        self.concept_2 = MagicMock(
+            spec=ModelDtsObject.ModelConcept,
+            qname=MagicMock(
+                namespaceURI='http://fasb.org/us-gaap/2015-01-31',
+                localName='NoncurrentAssets'
+            )
+        )
+        self.rel_1 = MagicMock(
+            spec=ModelRelationshipSet,
+            fromModelObject=self.concept_1,
+            toModelObject=self.concept_2,
+            arcrole='summation-item'
+        )
+        self.rel_set_1 = MagicMock(
+            spec=ModelRelationshipSet,
+            modelRelationships=[self.rel_1]
+        )
+
+        self.mock_ModelXbrlrelationshipSet = MagicMock(
+            spec=ModelRelationshipSet
+        )
+
+        self.mock_manager = MagicMock(
+            spec=ModelManager
+        )
+        self.mock_modelxbrl = MagicMock(
+            modelManager=self.mock_manager,
+            spec=ModelXbrl
+            )
+        self.mock_modelxbrl.relationshipSet.return_value = self.rel_set_1
+        self.mock_value = MagicMock(
+            spec=ValidateXbrl,
+            modelXbrl=self.mock_modelxbrl
+        )
+        pass
+
+    def tearDown(self):
+        pass
```

Comment:
teardown not needed if there are no contents, though that being said it is a perfect time to delete all the self. items you set up in the setUp function with `del self.concept_1`...etc.